functions/letters.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * Skrypt zawiera kilka ćwiczeniowych funkcji operujących na stringach.
+ *
+ * Przykład użycia:
+ * 1. Zdefiniuj zmienną $stringToTest na ciąg znaków, który chcesz przetestować.
+ * 2. Wywołuj kolejne funkcje zgodnie z ich dokumentacją.
+ * 3. Uruchom skrypt w konsoli: php -f letters.php
+ */
+
+declare(strict_types=1);
+
+/** this will clear the terminal */
+system('clear');
+
+
+/**
+ * now you can add your own word. Trim to remove enter key.
+ * @var $stringToTest string
+ */
+$stringToTest = trim(readline("Wpisz dowolne słowo: "));
+
+/**
+ * it's an array of variable types, to check what article should echo return
+ * @var $types array
+ */
+$types = ['object', 'array', 'integer'];
+
+
+// this condition is checking what article should be used.
+in_array(gettype($stringToTest), $types) ? $a = "an" : $a = "a";
```

Comment:
Zmienna **$a** jest niebezpieczna, bo nic nie mówi. Stosujmy opisowe nazwy zmiennych np. **$article**.

Reply:
W przypadku `in_array`, używajmy sprawdzenia typów (**strict type**), czyli ustawiajmy 3 parametr na _true_ zgodnie z:
http://php.net/manual/en/function.in-array.php

Reply:
Ogólnie polecam zapis krótszy:
`$a = in_array(gettype($stringToTest), $types, true) ? "an" : "a";`